### PR TITLE
Support #hide and #show for `para`

### PR DESCRIPTION
**Pull request by Schwad**
This was fun to get back into writing scarpe features!

Run the new example below to see it work. 

First tried to implement hide in the webview display service, but realized we can implement this at the DSL level (and probably should!).

Kinda wanted to implement a hide for all widgets but might need to do it for each one. No biggie. 

Anyways, some new functionality! enjoy!

### CODE REVIEW

File: examples/para/hide_and_show.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Shoes.app do
+  @banner = banner 'This is a big banner!'
```
Comment (DRBragg) [inline examples/para/hide_and_show.rb:2]: ⛏️ Be consistent with single or double quotes

**Comment by noahgibbs** [comment]
Hm. So you're not using HTML to hide stuff, you're taking the text items out of the normal variable and putting it into hidden.

So if we added text items, would they show up even though the para was hidden?

A hide-for-all-items should be possible. If you're going to add a field, maybe add a "hidden" property on widget?

File: lib/scarpe/para.rb [hunk 2]
```diff
@@ -35,6 +36,22 @@ def replace(*children)
       # This should signal the display widget to change
       self.text_items = text_children_to_items(@text_children)
     end
+
+    def hide
+      # idempotent
```
Comment (DRBragg) [inline lib/scarpe/para.rb:41]: 👍

**Review by DRBragg — APPROVED** [review]
🚀🚀🚀

**Comment by Schwad** [comment]
hidden: true is a lesser used but still canonical part of the API I'll be adding in in future. 

I'd love for that to be widget level. 

Since we needed the top level methods hide and show I couldn't add those in webview para as we're handling an instance of Scarpe::Para at runtime I think?

Anyways I'd tried.

If folks have ideas on implementing a global hide and show, and hidden: on all webview widgets I would accept that and revert this.

It is nice to work with the raw ruby here though...

**Comment by Schwad** [comment]
@noahgibbs I am conscious this may not be the final or perfect solution. Part of my hope here is to kind of get the creative juices flowing again as we ramp up for hopefully a more active June/July/August dev cycle.

And I find things like this helpful :)

**Comment by noahgibbs** [comment]
Sounds good. And I shouldn't be annoying about letting a potential future bug keep us from extending support.

**Review by noahgibbs — APPROVED** [review]

**Comment by Schwad** [comment]
@noahgibbs this adds support for hidden: true though! 

https://github.com/scarpe-team/scarpe/pull/179/commits/73c9f4eafce6803a7330d2237fffc543fd336509

**Pull request merged**